tests/examples/test_example_bots_training_data.py

```diff
@@ -80,8 +79,7 @@ def test_example_bot_training_on_initial_project(tmp_path: Path):
         importer.get_nlu_data()
         importer.get_stories()
 
-    # 2 for slot auto-fill removal warning
-    assert len(record) == 2
+    assert len(record) == 13
```

Comment:
@joejuzl For some reason that I don't quite understand, after yesterday's commits, the number of slot auto-fill removal `UserWarnings` have increased, I suppose it's because I'm using `Domain.from_dict()` in `Domain.empty`? Let me know if these test amendments are fine.

Reply:
13 sounds like a lot. Can you run the test and see if the warnings actually make sense?

Reply:
Yeah, I think they appear because we're instantiating domains with `from_dict` in many more places where we were using the `cls` constructor (for example, in `importers.py`) and that slot auto-fill removal warning appears only because of the slot validation that happens in `from_dict` method.

Reply:
Ok, well we don't want duplicate warnings, so we need to find a way to avoid that.

Reply:
What about this solution: in `from_dict`, add an if statement to run `validate_slot_mappings` only if there are slots defined in the domain, this way we avoid raising unnecessary warnings?